akka-actor-typed/src/main/scala/akka/actor/typed/eventstream/EventStream.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.eventstream
+
+import akka.actor.typed.internal.eventstream.SystemEventStream
+import akka.actor.typed._
+import akka.actor.typed.scaladsl.adapter._
+import akka.annotation.InternalApi
+
+import scala.reflect.ClassTag
+
+/**
+ * INTERNAL API
+ *
+ * Exposes a typed actor that interacts with the [[akka.actor.ActorSystem.eventStream]].
+ *
+ * It is used as an extension to ensure a single instance per actor system.
+ */
+@InternalApi private[akka] class EventStream(actorSystem: ActorSystem[_]) extends Extension {
+  val ref: ActorRef[EventStream.Command] =
+    actorSystem.internalSystemActorOf(SystemEventStream.behavior, "eventstream", Props.empty)
+}
+
+object EventStream extends ExtensionId[EventStream] {
+
+  sealed trait Command
+  case class Publish[E](event: E) extends Command
+  case class Subscribe[E](subscriber: ActorRef[E])(implicit classTag: ClassTag[E]) extends Command {
+    private[akka] def topic: Class[_] = classTag.runtimeClass
+  }
+
+  case class Unsubscribe[E](subscriber: ActorRef[E]) extends Command
```

Comment:
Make the messages final and add some scaladoc describing what they do, what happens if you subscribe/unsubscribe multiple times, isn't subscribed when unsubscribing.

We also need a factory method for Java users, at least for the subscribe since they cannot use `ClassTag`, and a test that makes sure that is nicely reachable from Java.

Reply:
Hm, to be honest I don't know what happens if you subscribe/unsubscribe multiple times, isn't subscribed when unsubscribing, I can't find it for the actual EventStream either in the docs. Should I just put a message that the mechanics are the same as the untyped event stream? 

Another thing that I have a question about is that subscribe returns a boolean but with the current implementation I have so far, it's fire and forget. Should we have some sort of SubscribeAck like in the distributed pub sub?

Reply:
Ok, I think a future ticket would be better, I can even continue this work at a later point :)

Reply:
https://github.com/akka/akka/issues/26854